### ISSUE TITLE
Pull-components stopped working

### ISSUE THREAD

**Issue by AndreaPontrandolfo** [storyblok/storyblok-cli#131]
**Current behavior:**
 Hi, welcome to the Storyblok CLI
 - Executing pull-components task
 X An error ocurred in pull-components task when load components data
 X An error occurred when executing the pull-components task: Error: [object Object]

**Expected behavior:**
Pull-components should work just fine. It was working last week. No changes made.

**Steps to reproduce:**
Just run storyblok pull-components --space <mySpace>

**Comment by angelikatyborska** [comment]
Hi 👋 I'm leaving this comment as a fellow user of the CLI, not as a maintainer (I'm clarifying this because technically I am part of the @storyblok github org).

I experience the same problem quite often. Whenever this happens, I run `storyblok login` to log in again, and then pulling components works again. I must be getting logged out, but I'm not sure why.

**Comment by AndreaPontrandolfo** [comment]
Actually, just re-logging didn't help. I had to actively logout and then login again to fix the issue.

**Comment by h3llr4iser** [comment]
I've also had it stop working for me. I've tried logging in again, but I'm still getting the same error:

```Hi, welcome to the Storyblok CLI

- Executing pull-components task
X An error ocurred in pull-components task when load components data
X An error occurred when executing the pull-components task: Error: [object Object]
```

**Comment by alvarosabu** [comment]
Hi @AndreaPontrandolfo @h3llr4iser unfortunately the error handling on this version of CLI is not really helpful for debugging purposes. Could you check a couple of things for me?

- if you are using Linux/OSX open `~/.netrc` can you confirm for me if you see this content:

```
machine api.storyblok.com
    login youremailaccount@storyblok.com
    password <YOUR_PERSONAL_ACCESS_TOKEN>
    region eu
```
- Could you logout explicitly with the `logout` command and then try again `login`
- Could you generate a new personal access token and try again?
- If all that fails, please get in touch with me via DM here in discord https://discord.gg/tgbZS9D5 with the spaceID you are using for the command 

Thanks

**Comment by alvarosabu** [comment]
> Hi 👋 I'm leaving this comment as a fellow user of the CLI, not as a maintainer (I'm clarifying this because technically I am part of the @storyblok github org).
> 
> I experience the same problem quite often. Whenever this happens, I run `storyblok login` to log in again, and then pulling components works again. I must be getting logged out, but I'm not sure why.

Yes it's a common issue of the CLI, sometimes the netrc file gets corrupted and the only way of cleaning it is with the `logout` command

**Comment by h3llr4iser** [comment]
I had the specified content in the ~/.netrc file. After trying the logout and logging back in, everything is working correctly now. Thanks!

**Comment by alvarosabu** [comment]
Awesome thanks for confirming, just FYI @h3llr4iser @AndreaPontrandolfo I'm currently working on the refactor of the CLI which will include better error handling in the near future, hopefully, errors like this will be easier to detect and less frequent 🙏🏻.

Thanks for your patience

**Comment by angelikatyborska** [comment]
@alvarosabu I got some more information regarding this bug, at least the way if happens to me.

> Sometimes the netrc file gets corrupted and the only way of cleaning it is with the logout command

That is not true in my case, the `.netrc` file is readable and contains the same data as it did before the error starts happening.

The error happened again to me today. I did some some debugging to confirm that the CLI reads the file, what contents it reads and so on. I discovered that the token stored in `.netrc` is a JWT that contains a timestamp. The timestamp that I found there this morning was `1733733783`, which is Monday, December 9, 2024 8:43:03 AM GMT. My attempts to use the pull command that failed were *after* that time. When I logged out and in again, the token changed and it now contains the timestamp 1 week from now.

Based on this, **I am guessing that the token** generated via `api.stoyblok.com/v1/users/login` (used when logging in via the CLI via email) **is only valid for 1 week**, so I will see this error every week and will need to log out and in again every week. 

If that is true, then this feature request would be a good idea to implement: https://github.com/storyblok/storyblok-cli/issues/77

<img width="1226" alt="SCR-20241209-mzeb" src="https://github.com/user-attachments/assets/0b3a5d3c-3920-41be-b45b-1ceb7a373bfa">
